free_fleet/src/dds_utils/DDSSubscribeHandler.hpp

```diff
@@ -116,7 +116,7 @@ class DDSSubscribeHandler
     {
       for (size_t i = 0; i < MaxSamplesNum; ++i)
       {
-        if (infos[i].valid_data)
+        if (infos[i].valid_data == true)
```

Comment:
I just saw https://github.com/eclipse-cyclonedds/cyclonedds/issues/1597, I'm curious whether the more accurate fix would be
```suggestion
        if (samples[i] != NULL)
```

Could you try that out with your scenarios too?

Reply:
I've tried your solution and I'm getting the same error. I don't have time to check it with gdb but probably is the same problem.

Reply:
gotcha, thanks for trying!